### ISSUE TITLE
Permission issues for host user when executing "docker-compose up"

### ISSUE THREAD

**Issue by alisamie97** [PrestaShop/PrestaShop#34410]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When I use the docker container setup, it's all good except for the time it wants to build the frontend and backend app with node and composer.
It shows permission errors and states that it could not install node and composer packages.

What did I do?

1. I made a new folder in my Ubuntu and cloned PrestaShop in that folder:
`git clone git@github.com:PrestaShop/PrestaShop.git myproject`

2. Then I started the docker containers as mentioned in the readme file:
`cd myproject && docker-compose up`

3. I get the errors at the last steps.

How I fix it?

1. I made a new folder in my Ubuntu and cloned PrestaShop in that folder:
`git clone git@github.com:PrestaShop/PrestaShop.git myproject`

2. I ran this command to fix permissions:
`sudo chmod -R 777 ./myproject`

3. Then I started the docker containers as mentioned in the readme file:
`cd myproject && docker-compose up`

I don't know if I have to do this manually or if it is supposed to be done automatically by docker, but seems like my host user does not have access to modify the new files and folders, when I change the access it works.
What could be wrong?

### Steps to reproduce

1. Clone the repo in a new folder:
`git clone git@github.com:PrestaShop/PrestaShop.git myproject`

2. Change directory to that folder and start docker containers:
`cd myproject && docker-compose up`

3. Check the errors in terminal when it installs node and composer packages

### PrestaShop version(s) where the bug happened

latest

**Comment by alisamie97** [comment]
Here are the errors that I get in the terminal:
Instead of `myproject` I used `beisatdev` as the folder name
```
beisatdev-prestashop-git-1  | * Launching the installer script...
beisatdev-prestashop-git-1  | Error : please install <a href="https://getcomposer.org/">composer</a>. Then run "php composer.phar install"
beisatdev-prestashop-git-1  | * Almost ! Starting web server now
beisatdev-prestashop-git-1  | 
beisatdev-prestashop-git-1  | AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 127.0.0.1. Set the 'ServerName' directive globally to suppress this message
beisatdev-prestashop-git-1  | AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 127.0.0.1. Set the 'ServerName' directive globally to suppress this message
beisatdev-prestashop-git-1  | [Sun Oct 29 09:22:32.024987 2023] [mpm_prefork:notice] [pid 1] AH00163: Apache/2.4.57 (Debian) configured -- resuming normal operations
beisatdev-prestashop-git-1  | [Sun Oct 29 09:22:32.025042 2023] [core:notice] [pid 1] AH00094: Command line: 'apache2 -D FOREGROUND'
beisatdev-prestashop-git-1  | npm ERR! code EACCES
beisatdev-prestashop-git-1  | npm ERR! syscall mkdir
beisatdev-prestashop-git-1  | npm ERR! path /var/www/html/themes/node_modules
beisatdev-prestashop-git-1  | npm ERR! errno -13
beisatdev-prestashop-git-1  | npm ERR! Error: EACCES: permission denied, mkdir '/var/www/html/themes/node_modules'
beisatdev-prestashop-git-1  | npm ERR!  [Error: EACCES: permission denied, mkdir '/var/www/html/themes/node_modules'] {
beisatdev-prestashop-git-1  | npm ERR!   errno: -13,
beisatdev-prestashop-git-1  | npm ERR!   code: 'EACCES',
beisatdev-prestashop-git-1  | npm ERR!   syscall: 'mkdir',
beisatdev-prestashop-git-1  | npm ERR!   path: '/var/www/html/themes/node_modules'
beisatdev-prestashop-git-1  | npm ERR! }
beisatdev-prestashop-git-1  | npm ERR! 
beisatdev-prestashop-git-1  | npm ERR! The operation was rejected by your operating system.
beisatdev-prestashop-git-1  | npm ERR! It is likely you do not have the permissions to access this file as the current user
beisatdev-prestashop-git-1  | npm ERR! 
beisatdev-prestashop-git-1  | npm ERR! If you believe this might be a permissions issue, please double-check the
beisatdev-prestashop-git-1  | npm ERR! permissions of the file and its containing directories, or try running
beisatdev-prestashop-git-1  | npm ERR! the command again as root/Administrator.
beisatdev-prestashop-git-1  | 
beisatdev-prestashop-git-1  | npm ERR! A complete log of this run can be found in:
beisatdev-prestashop-git-1  | npm ERR!     /var/www/.npm/_logs/2023-10-29T09_22_32_100Z-debug-0.log
beisatdev-prestashop-git-1  | npm ERR! code EACCES
beisatdev-prestashop-git-1  | npm ERR! syscall mkdir
beisatdev-prestashop-git-1  | npm ERR! path /var/www/html/admin-dev/themes/default/node_modules
beisatdev-prestashop-git-1  | npm ERR! errno -13
beisatdev-prestashop-git-1  | npm ERR! Error: EACCES: permission denied, mkdir '/var/www/html/admin-dev/themes/default/node_modules'
beisatdev-prestashop-git-1  | npm ERR!  [Error: EACCES: permission denied, mkdir '/var/www/html/admin-dev/themes/default/node_modules'] {
beisatdev-prestashop-git-1  | npm ERR!   errno: -13,
beisatdev-prestashop-git-1  | npm ERR!   code: 'EACCES',
beisatdev-prestashop-git-1  | npm ERR!   syscall: 'mkdir',
beisatdev-prestashop-git-1  | npm ERR!   path: '/var/www/html/admin-dev/themes/default/node_modules'
beisatdev-prestashop-git-1  | npm ERR! }
beisatdev-prestashop-git-1  | npm ERR! 
beisatdev-prestashop-git-1  | npm ERR! The operation was rejected by your operating system.
beisatdev-prestashop-git-1  | npm ERR! It is likely you do not have the permissions to access this file as the current user
beisatdev-prestashop-git-1  | npm ERR! 
beisatdev-prestashop-git-1  | npm ERR! If you believe this might be a permissions issue, please double-check the
beisatdev-prestashop-git-1  | npm ERR! permissions of the file and its containing directories, or try running
beisatdev-prestashop-git-1  | npm ERR! the command again as root/Administrator.
beisatdev-prestashop-git-1  | 
beisatdev-prestashop-git-1  | npm ERR! A complete log of this run can be found in:
beisatdev-prestashop-git-1  | npm ERR!     /var/www/.npm/_logs/2023-10-29T09_22_32_088Z-debug-0.log
beisatdev-prestashop-git-1  | npm ERR! code EACCES
beisatdev-prestashop-git-1  | npm ERR! syscall mkdir
beisatdev-prestashop-git-1  | npm ERR! path /var/www/html/admin-dev/themes/new-theme/node_modules
beisatdev-prestashop-git-1  | npm ERR! errno -13
beisatdev-prestashop-git-1  | npm ERR! Error: EACCES: permission denied, mkdir '/var/www/html/admin-dev/themes/new-theme/node_modules'
beisatdev-prestashop-git-1  | npm ERR!  [Error: EACCES: permission denied, mkdir '/var/www/html/admin-dev/themes/new-theme/node_modules'] {
beisatdev-prestashop-git-1  | npm ERR!   errno: -13,
beisatdev-prestashop-git-1  | npm ERR!   code: 'EACCES',
beisatdev-prestashop-git-1  | npm ERR!   syscall: 'mkdir',
beisatdev-prestashop-git-1  | npm ERR!   path: '/var/www/html/admin-dev/themes/new-theme/node_modules'
beisatdev-prestashop-git-1  | npm ERR! }
beisatdev-prestashop-git-1  | npm ERR! 
beisatdev-prestashop-git-1  | npm ERR! The operation was rejected by your operating system.
beisatdev-prestashop-git-1  | npm ERR! It is likely you do not have the permissions to access this file as the current user
beisatdev-prestashop-git-1  | npm ERR! 
beisatdev-prestashop-git-1  | npm ERR! If you believe this might be a permissions issue, please double-check the
beisatdev-prestashop-git-1  | npm ERR! permissions of the file and its containing directories, or try running
beisatdev-prestashop-git-1  | npm ERR! the command again as root/Administrator.
beisatdev-prestashop-git-1  | 
beisatdev-prestashop-git-1  | npm ERR! A complete log of this run can be found in:
beisatdev-prestashop-git-1  | npm ERR!     /var/www/.npm/_logs/2023-10-29T09_22_32_099Z-debug-0.log
```

**Comment by alisamie97** [comment]
I even changed the `docker-compose.yml` file from this:
```
volumes:
      - ./:/var/www/html:delegated
```
To:
```
volumes:
      - ./:/var/www/html:default
```

I searched for it and found this https://stackoverflow.com/a/63437557/5864034

So I thought that the `delegated` flag is the issue

**Comment by alisamie97** [comment]
here is the question just like my issue in stackoverflow: https://stackoverflow.com/questions/70725514/building-prestashop-using-docker-compose-error-eacces-permission-denied-acces

**Comment by alisamie97** [comment]
I found a workaround but I am not sure that everyone in PrestaShop company does this everytime:
Simply I allowed other users access this file, which means the user inside the container can access this bounded volume, at least I guess. I'm a newbie in Docker, forgive me **;)**
Before:
![image](https://github.com/PrestaShop/PrestaShop/assets/23106000/08521b1f-c5b7-4b5c-9107-ba09b24a9aad)
After:
![image](https://github.com/PrestaShop/PrestaShop/assets/23106000/d379aaae-1aaa-42f7-b064-0573791ac58b)

**Comment by alisamie97** [comment]
Now all the node and composer packages are installed. In other words, `themes/node_modules` and `vendor` are populated.
However, the new files generated were not allowed to be accessed by "Other" users, so I had to repeat things in my last comment again: https://github.com/PrestaShop/PrestaShop/issues/34410#issuecomment-1784055669
Then I could successfully see the installation page at this address: http://localhost:11400/install-dev/
I think now if any new file is added to this project I have to allow others to have access. Clearly what I did here is not a solution. What should I do?
